integreat_compass/cms/views/index.py

```diff
@@ -1,9 +1,125 @@
+
+        :param args: The supplied arguments
+        :type args: list
+
+        :param kwargs: The supplied keyword arguments
+        :type kwargs: dict
+
+        :return: The rendered template response
+        :rtype: ~django.template.response.TemplateResponse
+        """
+        offers = Offer.objects.all()
+        tags = Tag.objects.all()
+        list_of_offers = []
+        for offer in offers:
+            list_of_offers.append(self.store_offer_data_in_dict(offer))
+        amount_of_offers = len(list_of_offers)
+
+        return render(
+            request,
+            self.template_name,
+            {
+                "tags": tags,
+                "offers": list_of_offers,
+                "amount_of_offers": amount_of_offers,
+                "offer_group_types": offer_group_types.CHOICES,
+                "offer_mode_types": offer_mode_types.CHOICES,
+            },
+        )
+
+    def post(self, request):
+        """
+        This functions filters
+
+        :param request: The current request
+        :type request: ~django.http.HttpRequest
+
+        :return: The rendered template response
+        :rtype: ~django.template.response.TemplateResponse
+        """
+        offers = Offer.objects.all()
+        tags = Tag.objects.all()
+
+        selected_filters = {
+            "tags": request.POST.getlist("offer-tags"),
+            "mode_type": request.POST.get("mode-type"),
+            "group_type": request.POST.get("group-type"),
+            "search": request.POST.get("offer-search"),
+        }
+
+        filtered_offers = []
+        selected_filters["tags"] = [literal_eval(i) for i in selected_filters["tags"]]
+
+        for offer in offers:
+            tag_ids = [tag.id for tag in offer.tags.all()]
+
+            if any(item in selected_filters["tags"] for item in tag_ids):
+                filtered_offers.append(self.store_offer_data_in_dict(offer))
+                continue
+
+            # if selected_filters["mode_type"] == offer.mode_type:
+            #    filtered_offers.append(self.store_offer_data_in_dict(offer))
+            #    continue
+
+            # if selected_filters["group_type"] == offer.group_type:
+            #    filtered_offers.append(self.store_offer_data_in_dict(offer))
+            #    continue
+
+            # if re.search(selected_filters["search"], offer.versions.first().title, re.IGNORECASE):
+            #    filtered_offers.append(self.store_offer_data_in_dict(offer))
+            #    continue
+
+        amount_of_offers = len(filtered_offers)
+        return render(
+            request,
+            self.template_name,
+            {
+                "offers": filtered_offers,
+                "tags": tags,
+                "amount_of_offers": amount_of_offers,
+                "offer_group_types": offer_group_types.CHOICES,
+                "offer_mode_types": offer_mode_types.CHOICES,
+            },
+        )
+
+    def store_offer_data_in_dict(self, offer):
```

Comment:
I don't really understand the purpose of this function. You just can access all of these fields directly in the template, I think if you're renaming the variables you just make it harder to get where a variable is coming from.
Also, accessing `offer.versions.first()` multiple times generates a lot of unnecessary queries, please add a new cached property to the offer model to remember this field the first time you're accessing it.

Reply:
This was a mindblow! Thanks ;) Is applied now